x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentBody,
+  EuiTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { SetAppSearchBreadcrumbs as SetBreadcrumbs } from '../../../shared/kibana_breadcrumbs';
+import { SendAppSearchTelemetry as SendTelemetry } from '../../../shared/telemetry';
+import { LicenseContext, ILicenseContext, hasPlatinumLicense } from '../../../shared/licensing';
+import { KibanaContext, IKibanaContext } from '../../../index';
+
+import EnginesIcon from '../../assets/engine.svg';
+import MetaEnginesIcon from '../../assets/meta_engine.svg';
+
+import { LoadingState, EmptyState, ErrorState } from '../empty_states';
+import { EngineOverviewHeader } from '../engine_overview_header';
+import { EngineTable } from './engine_table';
+
+import './engine_overview.scss';
+
+export const EngineOverview: React.FC = () => {
+  const { http } = useContext(KibanaContext) as IKibanaContext;
+  const { license } = useContext(LicenseContext) as ILicenseContext;
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasErrorConnecting, setHasErrorConnecting] = useState(false);
+
+  const [engines, setEngines] = useState([]);
+  const [enginesPage, setEnginesPage] = useState(1);
+  const [enginesTotal, setEnginesTotal] = useState(0);
+  const [metaEngines, setMetaEngines] = useState([]);
+  const [metaEnginesPage, setMetaEnginesPage] = useState(1);
+  const [metaEnginesTotal, setMetaEnginesTotal] = useState(0);
+
+  const getEnginesData = async ({ type, pageIndex }: IGetEnginesParams) => {
+    return await http.get('/api/app_search/engines', {
+      query: { type, pageIndex },
+    });
+  };
+  const setEnginesData = async (params: IGetEnginesParams, callbacks: ISetEnginesCallbacks) => {
+    try {
+      const response = await getEnginesData(params);
+
+      callbacks.setResults(response.results);
+      callbacks.setResultsTotal(response.meta.page.total_results);
+
+      setIsLoading(false);
+    } catch (error) {
+      setHasErrorConnecting(true);
```

Comment:
For debugging purposes, I wonder if it would be a good idea to log the error here, otherwise the details will just be swallowed, right?

Reply:
Do you need to call `setIsLoading(false);` here as well? Or in a finally?

Reply:
The errors will get logged to the server for admins to inspect. I've deliberately avoided logging errors to the console/client as Kibana's security best practices mentioned not to as it could potentially expose compromising stack traces (IIRC)

Reply:
Link is in a non-public team repo, but here's the snippet:

>Check for accidental reveal of sensitive information
>    The biggest culprit is errors which contain stack traces or other sensitive information which end up in the HTTP Response

Reply:
> Do you need to call setIsLoading(false); here as well? Or in a finally?

No, because `<ErrorState>` takes precedence over/before `<LoadingState>` (line 81). Additionally it's very likely that I'll refactor how Error State is displayed to a React Router `<Redirect>` at some point, at which case there will also be no need to unset the loading state.